### ISSUE TITLE
state as timestamp with device_class attribute

### ISSUE THREAD

**Issue by eavanvalkenburg** [pippyn/Home-Assistant-Sensor-Afvalbeheer#52]
Hi, I was trying to get these look nice in a Home Feed card and that works only if the actual state is a timestamp and the device has a device_class: timestamp as attribute, then my visual can calculate time until etc. I made the change myself, by adding `CONF_DEVICE_CLASS, DEVICE_CLASS_TIMESTAMP` from const as imports, and adding `CONF_DEVICE_CLASS: DEVICE_CLASS_TIMESTAMP, ` in the attributes of the WasteSensor, and finally disabling all Today/Tomorrow logic in the update and just having this as line 575: `self._state = pick_update` and disabling the `if` in the line before and the whole `else` block in the lines after.

Could you consider making behaviour possible either by default or as an option. Visually this creates this, when using visuals that parse datetimes: 
![image](https://user-images.githubusercontent.com/13749212/79108067-d667e480-7d75-11ea-9a25-e8328b07deb2.png)

**Comment by eavanvalkenburg** [comment]
also checked the in-built [entity card](https://www.home-assistant.io/lovelace/entities/) and that would look like this and the format can be adapted to your needs:
![image](https://user-images.githubusercontent.com/13749212/79109878-8854e000-7d79-11ea-9cc8-fd65dc3ee590.png)

**Comment by pippyn** [comment]
There is already an option for this: https://github.com/pippyn/Home-Assistant-Sensor-Afvalbeheer#date-only. No need to change the code. 

And I'll look into adding the `CONF_DEVICE_CLASS`, but for now just add the `CONF_DEVICE_CLASS: DEVICE_CLASS_TIMESTAMP` with [a customization.](https://www.home-assistant.io/docs/configuration/customizing-devices/). It's quite easy, it would look something like:
```
customize:
    sensor.restafval:
        CONF_DEVICE_CLASS: DEVICE_CLASS_TIMESTAMP
```

**Comment by eavanvalkenburg** [comment]
missed that one, thanks, I do see a difference between and a datetime object as state:
![image](https://user-images.githubusercontent.com/13749212/79205986-f7483c80-7e3e-11ea-9d06-2c80fd447c62.png)

that is with `dateonly:1` and with the customization, so maybe a `dateobject: 1` setting?

**Comment by pippyn** [comment]
Sorry I don't know what you mean?

**Comment by eavanvalkenburg** [comment]
with the dateonly:1 setting and device_class customization, the state of these sensors is still a string and not a datetime object, some cards, including the Entities, can't apply their own formatting and display it correctly in that case and I prefer to let the front-end deal with formatting, rather than doing that in the integrations.

**Comment by pippyn** [comment]
I see.. Can you test a beta for me?
replace the contents of sensor.py with these contents:
https://raw.githubusercontent.com/pippyn/Home-Assistant-Sensor-Afvalbeheer/beta/custom_components/afvalbeheer/sensor.py
add `dateobject: 1` to your config and restart.

**Comment by eavanvalkenburg** [comment]
Just tested it, this worked for me:

Not sure what the logic should be, but I think if you want a dateobject, only then do you want the device_class set to timestamp, with date_only it is still a string and that get's parsed differently, and I also wouldn't expect them both to be set, so could be below or with an `or self.date_only == True`: 
```python
@property
def device_class(self):
  """Return the device class."""
  if self.date_object == True:
    return DEVICE_CLASS_TIMESTAMP
```

Replaced line 640 from:
```python
if self.date_only and date_diff >= 0:
```
to:
```python
if self.date_object:
   self._state = pick_update
elif self.date_only and date_diff >= 0:
```

**Comment by eavanvalkenburg** [comment]
and maybe it would be nice to add some hours to it as well to get the visuals looking even better, one of mine was this morning and now says 14 hours ago (midnight here) while it should maybe be 5 hours ago (9am?) or should be dependent on what the latest time is for your provider to put your cans outside?

**Comment by pippyn** [comment]
I already did add the check for date_only

I'm not going to add a time, because that might seem like it's information from the waste collector and it's not.

**Comment by eavanvalkenburg** [comment]
Not adding the time I understand. I commented the commit with the property, and it does still need the actual _state to be update with the datetime object instead of the string.

**Comment by pippyn** [comment]
You are right. When date_object is true, date_only should also be true. And the state should be the date without formatting. Changed the code

**Comment by eavanvalkenburg** [comment]
great, just dropped in that version, and works perfectly! Thanks @pippyn much appreciated!

**Comment by pippyn** [comment]
Released version 3.1.1 with state as dateobject implementation
